stdlib/public/core/CMakeLists.txt

```diff
@@ -424,6 +424,47 @@ add_swift_target_library(swiftCore
                     zippered
                  )
 
+# Backward deployment of Span on Apple platforms is implemented via the
+# libswiftCompatibility62.dylib shim.
+#
+# When a binary built with an older deployment target is run with a new
+# standard library, the symbols that were previously in the backward
+# compatibility shim are now in the standard library. To ensure the
+# dynamic linker continues to find these symbols, the compatibility
+# shim must be replaced by a symlink to the standard library.
+#
+# Since we're building the "new" standard library here, we create this
+# symlink in the same directory, from libswiftCompatibility62.dylib to
+# libswiftCore.dylib.
+#
+# The sources for the real libswiftCompatibility62.dylib are found in
+# stdlib/toolchain/Compatibility62. When running a binary with an old
+# standard library, this dynamic library must be present at run time.
+#
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
You should sync up with @etcwilde about also building this with the newer build system

Reply:
The compatibility libraries will be split out in the new build since they need to be built exactly as they were to match the original version. We don't want changes to the Swift or C++ version to impact them.